interview-questions.md

```diff
@@ -9,59 +9,61 @@ Then, research the question to expand on your answer. Even if you feel you have
 
-  Your answer:
-
-  Researched answer:
+  Your answer: `this` is the command/word/invocation that is ues to call an object in JavaScript.
 
+  Researched answer: The keyword `this` refers to the object that performs the current code section. It refers to the object that is performing the current function. If the reference function is a regular function, "this" refers to the global object. If the fuction is a method inside of an object, `this` refers to the object itself.
 
 
 2. What is React? Why would you use it?
 
-  Your answer:
+  Your answer: React is a framework (created by facebook) that allows websites to run and rerun code in an effeiacnt manner which does not required the user to constantly refresh the webpage they are viewing in order to see new content.  It is used because on its scalablity and efficiency.
 
-  Researched answer:
+  Researched answer: React is a front-end library, specifically focused on building user interfaces that are complex and responsive for web applications. If your project involves many components with different, often changing states – active/inactive navigation items, accordion sections expanded/collapsed, dynamic inputs, buttons active/disabled, user login and access permissions – then the project may be a good fit for React. React helps you manage those changing states and dynamically present different views to the user based on state information. Efficiency is one of React’s key points. A big part of its efficiency comes from React’s use of a Virtual DOM. React keeps a virtual copy of the page’s DOM behind the scenes. When a change occurs, React compares the current DOM with the incoming change being very careful to only update what has changed. This causes the browser to only re-render a small part of the page, instead of the entire thing. This makes React much more efficient and great for complex applications where changes occur frequently.
 
 
 
 3. Which lifecycle method is required in a React class component?
 
-  Your answer:
+  Your answer: I honestly wasn't able to recall the lifecycle methods off the top of my head.
 
-  Researched answer:
+  Researched answer:The render() method is required and will always be called, the others are optional and will be called if you define them.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
-  Your answer:
+  Your answer: JSX is essentually a hybrid of Javascript and HTML that is used in React.
+
+  Researched answer:JSX is a Javascript syntax extension used by ReactJS that can write Html-like Java scripts.  HTML is a hypertext language used to create items that appear on the screen, such as lists, divs, and images. JSX, on the other hand, is a JavaScript extension that allows you to create HTML in JavaScript code. The biggest difference between JSX and HTML is that nested JSX returns only one element. In other words, you still need a parent to create sibling items. It is not required for HTML.
```

Comment:
Awesome!  Because JSX only returns one element, we need to wrap multiple elements inside of react fragments.